src/Raven.Server/Web/System/TcpConnectionInfoHandler.cs

```diff
@@ -19,5 +27,116 @@ public Task Get()
 
             return Task.CompletedTask;
         }
+
+        [RavenAction("/info/tcp/pull", "GET", AuthorizationStatus.UnauthenticatedClients)]
+        public async Task GetPullReplicationInfo()
+        {
+            var databaseName = GetStringQueryString("databaseName");
+            var databaseGroupId = GetStringQueryString("databaseGroupId");
+            var definitionName = GetStringQueryString("definitionName");
+
+            var nodes = GetResponsibleNode(databaseName, databaseGroupId, definitionName);
+
+            // TODO: cache the tcp info for each node?
+            DynamicJsonArray output = new DynamicJsonArray();
+            var clusterTopology = ServerStore.GetClusterTopology();
+            var tasks = new List<Task<TcpConnectionInfo>>();
+            foreach (var node in nodes)
+            {
+                if (node == ServerStore.NodeTag)
+                {
+                    output.Add(Server.ServerStore.GetTcpInfoAndCertificates(HttpContext.Request.GetClientRequestedNodeUrl()));
+                }
+                else
+                {
+                    var nodeUrl = clusterTopology.GetUrlFromTag(node);
+                    tasks.Add(GetNodeTcpInfo(nodeUrl, databaseName));
+                }
+            }
+
+            await Task.WhenAll(tasks);
+            foreach (var task in tasks)
```

Comment:
Shouldn't this be in the proper order? Right now the current node tag is always first.

Reply:
Yes, _except_ that after that has been called, you call to `output.Add` directly, so it isn't in the right order